modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/system/info/item/provider/test/SystemObjectEntryInfoItemDetailsProviderTest.java

```diff
@@ -0,0 +1,198 @@
+		ServiceContextThreadLocal.popServiceContext();
+	}
+
+	@Test
+	public void testGetInfoItem() throws Exception {
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinition(
+				TestPropsValues.getCompanyId(), User.class.getSimpleName());
+
+		SystemObjectDefinitionManager systemObjectDefinitionManager =
+			_systemObjectDefinitionManagerRegistry.
+				getSystemObjectDefinitionManager(objectDefinition.getName());
+
+		long userId = systemObjectDefinitionManager.addBaseModel(
+			TestPropsValues.getUser(),
+			HashMapBuilder.<String, Object>put(
+				"alternateName", "john"
+			).put(
+				"emailAddress", "john@liferay.com"
+			).put(
+				"familyName", "Smith"
+			).put(
+				"givenName", "John"
+			).build());
+
+		String objectDefinitionItemClassName = StringBundler.concat(
+			objectDefinition.getClassName(), StringPool.POUND,
+			objectDefinition.getObjectDefinitionId());
+
+		InfoItemObjectProvider<SystemObjectEntry> infoItemObjectProvider =
+			_infoItemServiceRegistry.getFirstInfoItemService(
+				InfoItemObjectProvider.class, objectDefinitionItemClassName);
+
+		SystemObjectEntry systemObjectEntry =
+			infoItemObjectProvider.getInfoItem(
+				new ClassPKInfoItemIdentifier(userId));
+
+		InfoItemDetailsProvider<Object> infoItemDetailsProvider =
+			_infoItemServiceRegistry.getFirstInfoItemService(
+				InfoItemDetailsProvider.class, objectDefinitionItemClassName);
+
+		InfoItemDetails infoItemDetails =
+			infoItemDetailsProvider.getInfoItemDetails(systemObjectEntry);
+
+		Assert.assertEquals(
+			objectDefinitionItemClassName, infoItemDetails.getClassName());
+		Assert.assertEquals(
+			new InfoItemReference(
+				objectDefinitionItemClassName, systemObjectEntry.getClassPK()),
+			infoItemDetails.getInfoItemReference());
+
+		Map<String, Object> values = systemObjectEntry.getValues();
+
+		String externalReferenceCode = String.valueOf(
+			values.get("externalReferenceCode"));
+
+		systemObjectEntry = infoItemObjectProvider.getInfoItem(
+			_group.getGroupId(),
+			new ERCInfoItemIdentifier(
+				externalReferenceCode, _group.getExternalReferenceCode()));
+
+		infoItemDetails = infoItemDetailsProvider.getInfoItemDetails(
+			systemObjectEntry);
+
+		Assert.assertEquals(
+			objectDefinitionItemClassName, infoItemDetails.getClassName());
+		Assert.assertEquals(
+			new InfoItemReference(
+				objectDefinitionItemClassName, systemObjectEntry.getClassPK()),
+			infoItemDetails.getInfoItemReference());
+
+		infoItemDetails = infoItemDetailsProvider.getInfoItemDetails(
+			_group.getGroupId(), ERCInfoItemIdentifier.class,
+			systemObjectEntry);
+
+		Assert.assertEquals(
+			objectDefinitionItemClassName, infoItemDetails.getClassName());
+		Assert.assertEquals(
+			new InfoItemReference(
+				objectDefinitionItemClassName,
+				new ERCInfoItemIdentifier(
+					externalReferenceCode, _group.getExternalReferenceCode())),
+			infoItemDetails.getInfoItemReference());
```

Comment:
Since you used `_group.getGroupId()` in `infoItemDetailsProvider.getInfoItemDetails()` and it is the same `groupId` from the `systemObjectEntry` then the `scopeExternalReferenceCode` should be `null`.
(after you apply you the suggestions that I did)

Reply:
Add a scenario which you use `RandomTestUtil.randomLong()` in `infoItemDetailsProvider.getInfoItemDetails()`.

See `ObjectEntryFolderInfoItemDetailsProviderTest`